.github/CHANGELOG.md

```diff
@@ -2,6 +2,45 @@
 
 <h3>New features since last release</h3>
 
+* The `Spy()` context manager has been added, which mocks selected methods 
+  in PennyLane without executing them, but instead registers statistics 
+  about the calls. 
+  
+  At the moment, the class only mocks the device's `execute` method and registers the number 
+  of times it has been called. As opposed to a device's `num_executions` attribute, this 
+  allows a sneak peek of the circuits evaluated in an arbitrary computation.
```

Comment:
I had to read this a couple of times before I gathered the implication - maybe it's better to be explicit? E.g., 

> As opposed to a device's `num_executions` attribute, this  allows a sneak peek of the number of circuits evaluated in an arbitrary computation **without executing or running any quantum devices**.